serialize/src/lib.rs

```diff
@@ -241,3 +233,28 @@ pub fn buffer_bit_byte_size(modulus_bits: usize) -> (usize, usize) {
 pub const fn buffer_byte_size(modulus_bits: usize) -> usize {
     (modulus_bits + 7) / 8
 }
+
+pub fn batch_check_helper<'a, T>(
+    batch: impl Iterator<Item = &'a T>,
+) -> Result<(), SerializationError>
+where
+    T: 'a + Valid + Send + Clone,
+{
+    #[cfg(feature = "parallel")]
+    {
+        use ark_std::vec::Vec;
+        use rayon::prelude::{IntoParallelIterator, ParallelIterator};
+        batch
+            .map(|x| (*x).clone())
+            .collect::<Vec<T>>()
+            .into_par_iter()
+            .try_for_each(|e| e.check())?;
+    }
```

Comment:
Why does the `par_bridge` solution not work here?

Reply:
`ParallelBridge` is implemented for `Iterator + Send`. We no longer have `Send` bound on `batch` here. So we'd need to put an explicit bound: `batch: impl Iterator<Item = &'a T> + Send`, but this propagates all the way up to `Valid`, and would in turn lead to two definitions of `batch_check`:

```
    #[cfg(feature = "parallel")]
    fn batch_check<'a>(batch: impl Iterator<Item = &'a Self> + ParallelBridge)...

    #[cfg(not(feature = "parallel"))]
    fn batch_check<'a>(batch: impl Iterator<Item = &'a Self>) ...
```